paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetViewModelTest.kt

```diff
@@ -2390,6 +2393,78 @@ internal class PaymentSheetViewModelTest {
         assertThat(isPhoneRequired).isFalse()
     }
 
+    @Test
+    fun `Can complete payment after switching to another LPM from card selection with inline Link signup state`() =
+        runTest {
+            Dispatchers.setMain(testDispatcher)
+
+            val interceptor = spy(FakeIntentConfirmationInterceptor())
+
+            val viewModel = spy(
+                createViewModel(
+                    customer = EMPTY_CUSTOMER_STATE,
+                    intentConfirmationInterceptor = interceptor,
+                    linkState = LinkState(
+                        configuration = LINK_CONFIG,
+                        loginState = LinkState.LoginState.LoggedOut
+                    )
+                )
+            )
+
+            viewModel.primaryButtonUiState.test {
+                skipItems(1)
```

Comment:
skip seems not ideal. Can you do an assert so we can understand what's happening?